mule-extensions-api/src/main/java/org/mule/runtime/extension/api/runtime/config/ExtensionConfigurationManager.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.api.runtime.config;
+
+import org.mule.api.annotation.NoImplement;
+import org.mule.runtime.api.event.Event;
+import org.mule.runtime.api.meta.model.ComponentModel;
+import org.mule.runtime.api.meta.model.ExtensionModel;
+import org.mule.runtime.api.meta.model.config.ConfigurationModel;
+
+import java.util.Optional;
+
+/**
+ * This class is the access point to obtaining {@link ConfigurationInstance configuration instances} of the extensions in use.
+ * 
+ * @since 1.6
+ */
+@NoImplement
+public interface ExtensionConfigurationManager {
```

Comment:
This overlaps with methods in ExtensionManager. Why extract it here? Why not deprecated the ExtensionManager ones? How does this relate to split packages?